File: client-side/lib/domain/entities/ai_chat_session.dart
import 'package:equatable/equatable.dart';

class AIChatSession extends Equatable {
  final String id;
  final String title;
  final DateTime createdAt;
  final DateTime updatedAt;
  final String? model;

  const AIChatSession({
    required this.id,
    required this.title,
    required this.createdAt,
    required this.updatedAt,
    this.model,
  });

  @override
  List<Object?> get props => [id, title, createdAt, updatedAt, model];
}

File: internal/mappers/ai_chat_session.go
package mappers

import (
	"github.com/magomedcoder/legion/api/pb/aichatpb"
	"github.com/magomedcoder/legion/internal/domain"
)

func AIChatSessionToProto(session *domain.AIChatSession) *aichatpb.ChatSession {
	if session == nil {
		return nil
	}

	return &aichatpb.ChatSession{
		Id:        session.Id,
		Title:     session.Title,
		Model:     session.Model,
		CreatedAt: session.CreatedAt.Unix(),
		UpdatedAt: session.UpdatedAt.Unix(),
	}
}
